File: app/schemas/analytics.py
from decimal import Decimal

from pydantic import BaseModel


class RevenueTrendPoint(BaseModel):
    date: str
    revenue: Decimal


class TopProductItem(BaseModel):
    product_name: str
    sales_volume: int


class DashboardSummary(BaseModel):
    total_revenue: Decimal
    total_orders: int
    revenue_trend: list[RevenueTrendPoint]
    top_products: list[TopProductItem]

File: frontend/src/services/analytics.ts
import { request } from '../utils/request';

export interface RevenueTrendPoint {
  date: string;
  revenue: number | string;
}

export interface TopProductItem {
  product_name: string;
  sales_volume: number;
}

export interface DashboardSummary {
  total_revenue: number | string;
  total_orders: number;
  revenue_trend: RevenueTrendPoint[];
  top_products: TopProductItem[];
}

export async function getDashboardSummary(): Promise<DashboardSummary> {
  const response = await request.get<DashboardSummary>('/api/analytics/summary');
  return response.data;
}
